src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2347,43 +2363,70 @@ private IEnumerable<PSModuleInfo> CreateFakeModuleObject(IEnumerable<ModuleSpeci
             bool isConsideredCompatible = ModuleUtils.IsPSEditionCompatible(moduleManifestPath, inferredCompatiblePSEditions);
             if (!BaseSkipEditionCheck && !isConsideredCompatible)
             {
-                containedErrors = true;
-
-                if (writingErrors)
+                if (ExperimentalFeature.IsEnabled("PSWindowsPowerShellCompatibility"))
                 {
-                    message = StringUtil.Format(
-                        Modules.PSEditionNotSupported,
-                        moduleManifestPath,
-                        PSVersionInfo.PSEdition,
-                        string.Join(',', inferredCompatiblePSEditions));
+                    if (importingModule)
+                    {
+                        var commandInfo = new CmdletInfo("Import-Module", typeof(ImportModuleCommand));
```

Comment:
Should the cmdlet name be fully qualified?

Reply:
We use this syntax all over our code; also confirmed with @daxian-dbw that second parameter `implementingType` defines the implementation that is called.

Reply:
Another thing I thought about here was caching the CommandInfo object. Compared to the other operations here, it's not a huge performance drain, but it would remove some repeated code.